crates/sui-indexer/src/test_utils.rs

```diff
@@ -23,68 +23,71 @@ use crate::tempdb::get_available_port;
+    };
 
-    /// Instantiates a config for indexer in writer mode with the given snapshot config and epochs
-    /// to keep.
-    pub fn writer_mode(
-        snapshot_config: Option<SnapshotLagConfig>,
-        epochs_to_keep: Option<u64>,
-    ) -> Self {
-        Self::Writer {
-            snapshot_config: snapshot_config.unwrap_or_default(),
-            pruning_options: PruningOptions { epochs_to_keep },
-        }
-    }
+    println!("db_url: {db_url}");
+    println!("pool_config: {pool_config:?}");
+
+    let registry = prometheus::Registry::default();
+    init_metrics(&registry);
+
+    let pool = ConnectionPool::new(db_url.parse().unwrap(), pool_config)
+        .await
+        .unwrap();
+
+    let handle = {
+        let config = crate::config::JsonRpcConfig {
+            name_service_options: crate::config::NameServiceOptions::default(),
+            rpc_address: json_rpc_url.parse().unwrap(),
+            rpc_client_url: fullnode_url,
+        };
+        let token_clone = token.clone();
+        tokio::spawn(
+            async move { Indexer::start_reader(&config, &registry, pool, token_clone).await },
+        )
+    };
+
+    (handle, token)
 }
 
-pub async fn start_test_indexer(
+/// Wrapper over `Indexer::start_writer_with_config` to make it easier to configure an indexer
+/// writer for testing. If the config options are null, default values that have historically worked
+/// for testing will be used.
+pub async fn start_indexer_writer_for_testing(
     db_url: String,
-    rpc_url: String,
-    reader_writer_config: ReaderWriterConfig,
-    data_ingestion_path: PathBuf,
+    snapshot_config: Option<SnapshotLagConfig>,
+    pruning_options: Option<PruningOptions>,
+    data_ingestion_path: Option<PathBuf>,
+    cancel: Option<CancellationToken>,
 ) -> (
     PgIndexerStore,
     JoinHandle<Result<(), IndexerError>>,
     CancellationToken,
 ) {
-    let token = CancellationToken::new();
-    let (store, handle) = start_test_indexer_impl(
-        db_url,
-        rpc_url,
-        reader_writer_config,
-        Some(data_ingestion_path),
-        token.clone(),
-    )
-    .await;
-    (store, handle, token)
-}
+    let token = cancel.unwrap_or_default();
+    let snapshot_config = snapshot_config.unwrap_or(SnapshotLagConfig {
+        snapshot_min_lag: 5,
+        sleep_duration: 0,
+    });
+    let pruning_options = pruning_options.unwrap_or_default();
 
-/// Starts an indexer reader or writer for testing depending on the `reader_writer_config`.
-pub async fn start_test_indexer_impl(
-    db_url: String,
-    rpc_url: String,
-    reader_writer_config: ReaderWriterConfig,
-    data_ingestion_path: Option<PathBuf>,
-    cancel: CancellationToken,
-) -> (PgIndexerStore, JoinHandle<Result<(), IndexerError>>) {
-    // Reduce the connection pool size to 5 for testing
-    // to prevent maxing out
+    // Reduce the connection pool size to 10 for testing to prevent maxing out
```

Comment:
nit: the pool size is 5 in codes let's keep that?  prob from conflict resolution